exporter/observiqexporter/client_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observiqexporter
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+	"go.uber.org/zap"
+)
+
+// mockClock always reads the same provided time.
+// It also holds on to the function for AfterFunc, so it can be triggered at without relying on timing.
+type mockClock struct {
+	time        time.Time
+	timeoutFunc func()
+}
+
+func (mc mockClock) Now() time.Time {
+	return mc.time
+}
+
+/* Mock after func; Does not call the function, just returns a timer of the given duration */
+func (mc *mockClock) AfterFunc(d time.Duration, f func()) *time.Timer {
+	mc.timeoutFunc = f
+	return time.NewTimer(d)
+}
+
+func newMockClock(t time.Time) clock {
+	return &mockClock{time: t}
+}
+
+const testURL = "http://example.com"
+
+type testRoundTripper func(req *http.Request) *http.Response
+
+func (t testRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return t(req), nil
+}
+
+type testFailRoundTripper struct {
+	err error
+}
+
+func (t testFailRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return nil, t.err
+
+}
+
+func newTestHTTPClient(respCode *int, respBody *string, err error) *http.Client {
+	var rt http.RoundTripper = testRoundTripper(func(req *http.Request) *http.Response {
+		return &http.Response{
+			StatusCode: *respCode,
+			Body:       ioutil.NopCloser(bytes.NewBufferString(*respBody)),
+			Header:     make(http.Header),
+		}
+	})
+
+	if err != nil {
+		rt = testFailRoundTripper{err: err}
+	}
+
+	return &http.Client{
+		Transport: rt,
+	}
+}
+
+func newTestClient(config *Config, httpClient *http.Client, clock clock) *client {
+	return &client{
+		client: httpClient,
+		logger: zap.NewNop(),
+		config: config,
+		clock:  clock,
+	}
+}
+
+func createLogData(clock clock) pdata.Logs {
+	logs := pdata.NewLogs()
+	logs.ResourceLogs().Resize(1)
+
+	now := clock.Now()
```

Comment:
I think I get what you're doing here by passing in a clock, but I think there is another solution that is more typical.

Somewhere in the package: `var timeNow = time.Now`

In this method: `now := timeNow()`

In tests: `now := staticNow`

Reply:
Yeah, this solution (the clock interface) is what I came up when researching how people mock time in go; I've refactored to be more in line with your solution. I think timeNow isn't re-assigned in tests just yet, but it definitely will as `converter.go` gets filled out.